### DISCUSSION TITLE
Copilot extension failed to activate on VScode's remote workspace

### DISCUSSION THREAD

**Discussion by glass-panel** [community/community#31984] [Copilot Conversations] [answered]
### Description:
The Github Copilot extension on my vscode's remote workspace worked just fine before I updated my vscode to 1.70 also the Copilot extension. But now it just showing "Not yet activated." in the Runtime Status of its extension page, without any log in its output.
I've checked the Remote Extension Host's log and it shows that it do tried to activate the Copilot extension:
`[exthost] [info] ExtensionService#_doActivateExtension GitHub.copilot, startup: false, activationEvent: 'onStartupFinished'`. 

The only related log I can find is from VScode's Dev Tools termial: 
```
ERR Cannot read properties of undefined (reading 'emit'): TypeError: Cannot read properties of undefined (reading 'emit')
    at ClientRequest.<anonymous> (/home/{username}/.vscode-server/extensions/github.copilot-1.43.6621/dist/extension.js:8:504635)
    at Object.onceWrapper (node:events:646:26)
    at ClientRequest.emit (node:events:526:28)
    at Socket.socketOnEnd (node:_http_client:466:9)
    at Socket.emit (node:events:538:35)
    at endReadableNT (node:internal/streams/readable:1345:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

### My Environment:
 - VScode: 1.71.0
 - Copilot extension: v1.43.6621
 - Local OS: Windows 10 21H2
 - Remote worksapce OS: openSUSE Leap 15.3 by ssh

### What I have tried
- Logged out and in for my Github account
- Re-install the plugin both local and remote side.
- Delete remote ~/.vscode-server/ directory and set up remote connetion again

And none of them worked.

Thanks for your patience.

**Comment by philkuz** [comment]
I don't know if I'm running into the same problem as you and clearly there is a bug here, but here's a work around that I've used to reactivate copilot
1. Open extension page
3. Disable extension and reload window
4. Open extension page
5. Enable extension
6. Follow the sign-in flow from the toast that pops up in the bottom right.

**Comment by glass-panel** [comment, marked as answer]
It turns out that my http_proxy and https_proxy env used socks:// prefix and vscode's http_client API dosen't support that, causing this error.
